iml-services/iml-device/src/main.rs

```diff
@@ -86,31 +91,131 @@ async fn main() -> Result<(), ImlDeviceError> {
 
     tokio::spawn(server);
 
-    let mut s = consume_data::<Device>("rust_agent_device_rx");
+    let mut s = consume_data::<(Device, HashSet<Mount>)>("rust_agent_device_rx");
 
     let pool = iml_orm::pool().unwrap();
 
-    while let Some((f, d)) = s.try_next().await? {
+    while let Some((host, (devices, mounts))) = s.try_next().await? {
         let mut cache = cache2.lock().await;
 
-        cache.insert(f.clone(), d.clone());
+        cache.insert(host.clone(), devices.clone());
 
         let device_to_insert = NewChromaCoreDevice {
-            fqdn: f.to_string(),
-            devices: serde_json::to_value(d).expect("Could not convert incoming Devices to JSON."),
+            fqdn: host.to_string(),
+            devices: serde_json::to_value(devices)
+                .expect("Could not convert incoming Devices to JSON."),
         };
 
-        let new_device = diesel::insert_into(table)
+        let new_device = diesel::insert_into(dvc::table)
             .values(device_to_insert)
-            .on_conflict(fqdn)
+            .on_conflict(dvc::fqdn)
             .do_update()
-            .set(devices.eq(excluded(devices)))
+            .set(dvc::devices.eq(excluded(dvc::devices)))
             .get_result_async::<ChromaCoreDevice>(&pool)
             .await
             .expect("Error saving new device");
 
-        tracing::info!("Inserted device from host {}", new_device.fqdn);
-        tracing::trace!("Inserted device {:?}", new_device);
+        tracing::info!("Inserted device from host {}", host);
+        tracing::debug!("Inserted {:?}", new_device);
+
+        let lustre_mounts = mounts
+            .into_iter()
+            .filter(|m| m.fs_type.0.as_str() == "lustre")
+            .filter_map(|m| {
+                m.source
+                    .0
+                    .to_str()
+                    .and_then(|p| p.splitn(2, ":/").nth(1))
+                    .map(|fs| (fs.to_string(), m.target.0.to_str().map(String::from)))
+            })
+            .collect::<Vec<_>>();
+
+        tracing::debug!("Client mounts at {}: {:?}", host, &lustre_mounts);
+
+        let state_modified_at = chrono::offset::Utc::now().into_sql::<sql_types::Timestamptz>();
+        let not_deleted = Some(true).into_sql::<sql_types::Nullable<sql_types::Bool>>();
+        let cross = true.into_sql::<sql_types::Bool>();
+        let immutable_state = false.into_sql::<sql_types::Bool>();
+
+        let insert_mounts = lustre_mounts
+            .iter()
+            .map(|(fs, tg)| {
+                let mp = tg
+                    .clone()
+                    .into_sql::<sql_types::Nullable<sql_types::Text>>();
+                mgtfs::table
+                    .inner_join(mgthost::table.on(cross))
+                    .inner_join(djct::table.on(cross))
```

Comment:
```suggestion
                    .inner_join(djct)
```

There is a `joinable!` implemented for this in the schema.